### PR TITLE
Support batch update element properties with strategies

### PR DESCRIPTION
**Pull request by imbajin**
The strategy in JSON is case sensitive now, wait for jackthon update to V2.9+ to support case-insensitive original

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/loader/task/SingleInsertTask.java [hunk 1]
```diff
@@ -60,7 +61,13 @@ public void run() {
         LoadMetrics metrics = this.context().summary().metrics(this.struct());
         for (GE element : this.batch()) {
             try {
-                this.addSingle(type, element);
+                if (this.struct().updateStrategies().isEmpty()) {
+                    this.addSingle(type, element);
+                } else {
+                    // TODO: Adapt single update later
+                    this.updateBatch(type, ImmutableList.of(element),
+                                     options.checkVertex);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/task/SingleInsertTask.java]: add updateSingle() and move the two lines into updateSingle()

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java [hunk 2]
```diff
@@ -1563,4 +1563,36 @@ public void testParserNotThrowException() {
...
+                "--test-mode", "true"
+        };
+        HugeGraphLoader.main(args);
+
+        List<Vertex> vertices = CLIENT.graph().listVertices();
+        List<Edge> edges = CLIENT.graph().listEdges();
+
+        Assert.assertEquals(vertices.size(), 1);
+        Assert.assertEquals(vertices.get(0).property("age"), 18);
+        Assert.assertEquals(((List) vertices.get(0).property("set")).size(), 3);
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java]: add assert containsAll(ImmutableList.of(str1, str2, str3))

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java [hunk 3]
```diff
@@ -1563,4 +1563,36 @@ public void testParserNotThrowException() {
         };
         HugeGraphLoader.main(args);
     }
+
+    @Test
+    public void testBatchUpdateElement() {
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java]: add a test checkVertex=true|false

File: src/test/resources/update_by_strategy/struct.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{
+  "vertices": [
+    {
+      "label": "person",
+      "update_strategies": {
+        "age": "SMALLER",
+        "set": "UNION"
```
Comment (javeme) [inline src/test/resources/update_by_strategy/struct.json]: add a test for invalid strategy name

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java (hunk 2 above)
Comment (imbajin) [reply src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java]: The order of set is not fixed, so replace it (list is the default collection in json)

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java [hunk 5]
```diff
@@ -1595,4 +1596,53 @@ public void testBatchUpdateElement() {
...
+                "-s", configPath("update_by_strategy/schema.groovy"),
+                "-g", GRAPH,
+                "-h", SERVER,
+                "--num-threads", "2",
+                "--test-mode", "true"
+        };
+        
+        Assert.assertThrows(SerializeException.class, () -> {
+            HugeGraphLoader.main(args);
+        });
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java]: assert error message

File: src/main/java/com/baidu/hugegraph/loader/task/SingleInsertTask.java [hunk 6]
```diff
@@ -65,8 +65,7 @@ public void run() {
                     this.addSingle(type, element);
                 } else {
                     // TODO: Adapt single update later
-                    this.updateBatch(type, ImmutableList.of(element),
-                                     options.checkVertex);
+                    this.updateSingle(type, options, element);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/task/SingleInsertTask.java]: move TODO to line 91

**Pull request merged**